src/ChainState.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState } from "react";
+
+import { Grid, Form, Dropdown, Button, Input } from "semantic-ui-react";
+
+export default function Metadata(props) {
+  const { api } = props;
+
+  const [modulesList, setModulesList] = useState([]);
+  const [output, setOutput] = useState("");
+  const [storageItemsList, setStorageItemsList] = useState([]);
+
+  const initialState = {
+    module: "",
+    storageItem: "",
+    input: ""
+  };
+  const [formState, setFormState] = useState(initialState);
+  const { module, storageItem, input } = formState;
+
+  useEffect(() => {
+    let queries = api.query;
+    let modules = Object.keys(queries).map(module => ({
+      key: module,
+      value: module,
+      text: module
+    }));
+
+    setModulesList(modules);
+  }, [api]);
+
+  useEffect(() => {
+    if (module !== "") {
+      let queries = eval("api.query." + module + ".valueOf()");
+      let storageItems = Object.keys(queries).map(storage => ({
```

Comment:
Can just be `Object.keys(api.query[module])` (no eval)

Reply:
perfect, thank you